src/Symfony/Component/Scheduler/Messenger/TaskMessageHandler.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Scheduler\Messenger;
+
+use Cron\CronExpression;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Symfony\Component\Scheduler\Worker;
+
+/**
+ * @author Guillaume Loulier <contact@guillaumeloulier.fr>
+ */
+final class TaskMessageHandler implements MessageHandlerInterface
+{
+    /**
+     * @var Worker
+     */
+    private $worker;
+
+    public function __construct(Worker $worker)
+    {
+        $this->worker = $worker;
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function __invoke(TaskMessage $message): void
+    {
+        $task = $message->getTask();
+
+        if (!CronExpression::factory($task->get('expression'))->isDue(new \DateTimeImmutable())) {
```

Comment:
What if the messages took 1 hours to be proceed? Maybe `isDue` should take the date of creation of the task instead of `now`

Reply:
Improved